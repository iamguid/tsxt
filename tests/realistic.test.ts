import template, { ctx } from "../compiled-templates/realistic.template";

it("template generates correctly", () => {
  const result = template(ctx);
  expect(result).toMatchSnapshot();
});
