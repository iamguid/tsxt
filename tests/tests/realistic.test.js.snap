// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template generates correctly 1`] = `
"// "GENERATED CODE -- DO NOT EDIT!"

import * as grpcWeb from "grpc-web"
import * jspb from "google-protobuf"

import * a from "./a";
import * b from "./b";
import * c from "./c";

export interface ISomeClassName {
    methodA: (request: InputType, metadata: grpcWeb.Metadata) => grpcWeb.ClientReadableStream<OutputType>;
    methodB: (request: InputType, metadata: grpcWeb.Metadata) => Promise<OutputType>;
}

export class SomeClassName implements ISomeClassName {
    public methodA(request: InputType, metadata: grpcWeb.Metadata): grpcWeb.ClientReadableStream<OutputType> {
        return void;
    }
    public methodB(request: InputType, metadata: grpcWeb.Metadata): Promise<OutputType> {
        return void;
    }
}

"
`;
